### DISCUSSION TITLE
Assessment of the difficulty in porting CPU architecture for cereal

### DISCUSSION THREAD

**Discussion by wangyuliu** [USCiLab/cereal#802] [General]
Hello everyone! I am working on implementing a tool to assess the complexity of CPU architecture porting. It primarily focuses on RISC-V architecture porting. In fact, the tool may have an average estimate of various architecture porting efforts.My focus is on the overall workload and difficulty of transplantation in the past and future,even if a project has already been ported.As part of my dataset, I have collected the **cereal** project. **I would like to gather community opinions to support my assessment. I appreciate your help and response!** Based on scanning tools, the porting complexity is determined to be simple, with a small amount of code related to the CPU architecture in the project.  Is this assessment accurate?I look forward to your help and response.

**Comment by lordvictory** [comment]
Not sure exactly what CPU Architecture porting you're referring to.  AFAIK, the cereal source code is generic C++11 code, and likely runs correctly (once built) on any processor that has a C++11 toolchain.

I can state with 100% certainty, that when I build my generic C++11 application network code for Linux x86_64 as well as Linux ARM32 and Linux ARM64 (risc systems), I can freely interchange messages containing type instance information between said disparate systems using this scheme, regardless of CPU Architecture.

The messaging communications software marshalls custom C++ Type instances by first using cereal to flatten them into a message and sending it to a disparate system (e.g. a different processor architecture) over gigabit ethernet.  When the message arrives it's de-marshalled using cereal to recreate the C++ Type instance on the arriving system.  This scheme uses the portable binary encoding format of cereal. Any CPU Architecture differences are handled automatically and properly by cereal, AFAIK...I've not seen any issues in many hours of testing.

The tests conducted all used the same toolchain, GNU g++ v13.2.  Each toolchain was built for each platform independently.  NXP Yocto build system was used for the Embedded Targets, ARM32 and ARM64.  Ubuntu 24.04 was used for x86_64 Linux build targets.
